src/main/java/org/deri/tarql/CSVParser.java

```diff
@@ -76,10 +76,22 @@ public CSVParser(Reader reader, boolean varsFromHeader, Character delimiter, Cha
 	private Var toVar(String s) {
 		if (s == null)
 			return null;
-        s = s.trim().replace(" ", "_").replace("-", "_").replace("?", "_").replace("%", "_").replace("(", "_").replace(")", "_");
+		/* SPARQL 1.1 VAR Gramar ?
+		 VARNAME	  ::=  	( PN_CHARS_U | [0-9] ) ( PN_CHARS_U | [0-9] | #x00B7 | [#x0300-#x036F] | [#x203F-#x2040] )*
+		 PN_CHARS_U	  ::=  	PN_CHARS_BASE | '_'
+		 PN_CHARS_BASE	  ::=  	[A-Z] | [a-z] | [#x00C0-#x00D6] | [#x00D8-#x00F6] | [#x00F8-#x02FF] | [#x0370-#x037D] | [#x037F-#x1FFF] | [#x200C-#x200D] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
+			I've omitted UTF-16 character range #x10000-#xEFFFF.
+		*/
+
+		String PN_CHARS_BASE = "A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD";
+		String pattern =   PN_CHARS_BASE + "0-9\u00B7\u0300-\u036F\u203F-\u2040";
+
+        s = s.trim().replaceAll("[^" + pattern + "]", "_").replace(":", "");
```

Comment:
Please fix indentation

Reply:
took a while on my end... :D it is fixed now.